### PR TITLE
New rule: line-length-limit

### PR DESCRIPTION
**Pull request by xuri**

### CODE REVIEW

File: untyped.toml [hunk 1]
```diff
@@ -16,3 +16,6 @@
 [rule.range-val-in-closure]
 [rule.waitgroup-by-value]
 [rule.atomic]
+[rule.empty-lines]
+[rule.line-length-limit]
+    arguments = [200]
```
Comment (mgechev) [inline untyped.toml]:
```suggestion
    arguments = [200]

```

Comment (mgechev) [reply untyped.toml]: A trailing newline.

File: rule/line-length-limit.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+	if len(arguments) != 1 {
+		panic(`invalid configuration for "line-length-limit"`)
+	}
+
+	max, ok := arguments[0].(int64) // Alt. non panicking version
+	if !ok {
+		panic(fmt.Sprintf(`invalid value passed as return results number to the "line-length-limit" rule; need int64 but got %T`, arguments[0]))
+	}
+	if max < 0 {
+		panic(`the value passed as return results number to the "line-length-limit" rule cannot be negative`)
```
Comment (mgechev) [inline rule/line-length-limit.go]: Can we simplify the panic message?

**Comment by xuri** [comment]
@mgechev @chavacava Thanks for your advice, I have fixed it.

**Comment by chavacava** [comment]
It's looks good.

**Comment by harleenarora** [comment]
Hi, This rule is not working with exclude parameter. I have add a pattern in exclude but its still show the error. Here is my code 
name: line-length-limit
          disabled: false
          severity: warning
          arguments: [120]
          exclude:
             - '^type\s+[A-Z][a-zA-Z0-9_]*\s*\{'

**Comment by chavacava** [comment]
Hi @harleenarora, thanks for reporting the issue.

The rule hasn't an exclusion parameter so it's expected to not to take it into account. It seems you are using the rule through `golangci-lint`, you might open an issue there asking for help on how to use the exclusion patterns.

**Pull request merged**